src/server/gc_worker/gc_worker.rs

```diff
@@ -367,10 +369,18 @@ where
         MvccTxn::new(TimeStamp::zero(), concurrency_manager)
     }
 
-    fn flush_txn(txn: MvccTxn, limiter: &Limiter, engine: &E) -> Result<()> {
-        let write_size = txn.write_size();
-        let modifies = txn.into_modifies();
-        if !modifies.is_empty() {
+    fn flush_txn(
+        limiter: &Limiter,
+        engine: &E,
+        txns: HashMap<u64, MvccTxn>, // region id -> MvccTxn for this region
```

Comment:
How about using `&mut HashMap`? So it can be reused outside.

Reply:
`into_modifies` of MvccTxn will be used in this method.

Reply:
```
for (id, txn) in txns.drain() {
    txn.into_modifies()
}
```

Reply:
Good catch.

Reply:
Good suggestion.